contrib/go/src/python/pants/contrib/go/tasks/go_compile.py

```diff
@@ -23,9 +24,16 @@ class GoCompile(GoWorkspaceTask):
   """
 
   @classmethod
+  @memoized_method
+  def implementation_version(cls):
```

Comment:
Options from subsystem dependencies are included in a task's fingerprint.

It is a known bug that they aren't included _transitively_ through the subsystem dependency graph, but in this case it should work because the superclass of this task depends directly on that subsystem.